### ISSUE TITLE
Helm upgrade fails on kubernetes v1.25+ due to PodSecurityPolicy dependency

### ISSUE THREAD

**Issue by scottnimos** [grafana/k8s-monitoring-helm#390]
Hi Grafana Dev Team.

I'm unable to upgrade the grafana/k8s-monitoring Helm chart on Kubernetes v1.25.11. The upgrade process fails with an error indicating a dependency on the PodSecurityPolicy resource, which is no longer supported in Kubernetes 1.25+.

**Environment:**
Kubernetes Version: v1.25.11

**Steps to Reproduce:**

1. Add the Grafana Helm chart repository: helm repo add grafana https://grafana.github.io/helm-charts
2. Update Helm repositories: helm repo update
3. Attempt the upgrade: helm upgrade --install --atomic --timeout 300s grafana-k8s-monitoring grafana/k8s-monitoring -n grafana-agent --create-namespace -f values.yaml

**Expected Behavior:**
Helm chart to deploy successfully on Kubernetes versions where PodSecurityPolicies have been removed.  Please update the chart to address this incompatibility.

Cheers

**Comment by scottnimos** [comment]
closing posted in wrong repository.

Here is a the solution if anyone comes across this
https://github.com/grafana/helm-charts/blob/grafana-6.36.1/charts/grafana/templates/podsecuritypolicy.yaml
{{- if .Values.rbac.pspEnabled }}

**Comment by acmacalister** [comment]
@scottnimos, how did you modify that grafana chart before deploying this chart? Little confused on how to implement the solution for this.

**Comment by scottnimos** [comment]
Hi @acmacalister apologies, I was really confused the other day. Though that led me down that something was up on my end because the chart did have code to recognise if the PodSecurityPolicy is available. (confirmed from the link I provided)

After trying to uninstall grafana/k8s-montioring chart, helm spitted out uninstall error. I had to manually delete the resources on the cluster as helm no longer had a reference to a chart being installed. I then proceeded to install the chart again and it worked as expected.

**Comment by acmacalister** [comment]
@scottnimos no worries at all. I appreciate the reply. 

I ended up having to delete/uninstall the helm chart and reinstall as well.

tldr, this seem to only effect users upgrading their k8s-monitoring deployment as older version had `grafana-k8s-monitoring-prometheus-node-exporter` which still had definitions for PodSecurityPolicy resources, which were removed in k8 1.25+. Since the resource no longer exists, it will cause upgrades to fail. I tried this plugin:

https://github.com/helm/helm-mapkubeapis

`helm mapkubeapis grafana-k8s-monitoring`

and it successfully allowed me to do the upgrade, but kept timing out. 

`helm upgrade --install --atomic --timeout 300s grafana-k8s-monitoring grafana/k8s-monitoring --values values.yml`

I tried deleting the secrets as others suggested: 

`k delete secrets/sh.helm.release.v1.grafana-k8s-monitoring.v1`

but that still failed. I removed the install and just install again and that got everything kosher again.
